File: decompiled_melonDS/sources/me/magnum/melonds/ui/settings/fragments/GeneralPreferencesFragment.java
package me.magnum.melonds.ui.settings.fragments;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.PowerManager;
import androidx.fragment.app.b1;
import androidx.preference.Preference;
import androidx.preference.SwitchPreference;
import androidx.preference.o;
import androidx.preference.p;
import com.smp.masterswitchpreference.MasterSwitchPreference;
import kf.x1;
import me.magnum.melonds.R;
import yb.n;
/* compiled from: r8-map-id-68a076d6c8e19c01d0388d4729d627070c4f7cff1f6777b7d5e4497c81f45b39 */
/* loaded from: classes.dex */
public final class GeneralPreferencesFragment extends Hilt_GeneralPreferencesFragment implements ci.l {
    public final n Y = new n(new e(this, 2));
    public sd.i Z;

    /* renamed from: b0  reason: collision with root package name */
    public sd.d f9590b0;

    /* renamed from: c0  reason: collision with root package name */
    public x1 f9591c0;

    /* renamed from: d0  reason: collision with root package name */
    public MasterSwitchPreference f9592d0;

    /* renamed from: e0  reason: collision with root package name */
    public final h.c f9593e0;

    /* renamed from: f0  reason: collision with root package name */
    public final h.c f9594f0;

    public GeneralPreferencesFragment() {
        h.c registerForActivityResult = registerForActivityResult(new b1(1), new f(this, 0));
        registerForActivityResult.getClass();
        this.f9593e0 = registerForActivityResult;
        h.c registerForActivityResult2 = registerForActivityResult(new b1(1), new f(this, 1));
        registerForActivityResult2.getClass();
        this.f9594f0 = registerForActivityResult2;
    }

    @Override // ci.l
    public final String getTitle() {
        String string = getString(R.string.category_general);
        string.getClass();
        return string;
    }

    @Override // androidx.preference.a0
    public final void onCreatePreferences(Bundle bundle, String str) {
        setPreferencesFromResource(R.xml.pref_general, str);
        addPreferencesFromResource(R.xml.pref_general_updates);
        Preference findPreference = findPreference("enable_rewind");
        findPreference.getClass();
        this.f9592d0 = (MasterSwitchPreference) findPreference;
        Preference findPreference2 = findPreference("enable_sustained_performance");
        findPreference2.getClass();
        SwitchPreference switchPreference = (SwitchPreference) findPreference2;
        ci.k kVar = (ci.k) this.Y.getValue();
        MasterSwitchPreference masterSwitchPreference = this.f9592d0;
        if (masterSwitchPreference != null) {
            kVar.getClass();
            ci.k.a(masterSwitchPreference);
            Context requireContext = requireContext();
            requireContext.getClass();
            Object systemService = requireContext.getSystemService("power");
            systemService.getClass();
            switchPreference.setVisible(((PowerManager) systemService).isSustainedPerformanceModeSupported());
            Preference findPreference3 = findPreference("backup_settings");
            if (findPreference3 != null) {
                findPreference3.setOnPreferenceClickListener(new p(this) { // from class: me.magnum.melonds.ui.settings.fragments.g
                    public final /* synthetic */ GeneralPreferencesFragment B;

                    {
                        this.B = this;
                    }

                    @Override // androidx.preference.p
                    public final void a(Preference preference) {
                        switch (r2) {
                            case 0:
                                this.B.f9593e0.a(null);
                                return;
                            default:
                                this.B.f9594f0.a(null);
                                return;
                        }
                    }
                });
            }
            Preference findPreference4 = findPreference("restore_settings");
            if (findPreference4 != null) {
                findPreference4.setOnPreferenceClickListener(new p(this) { // from class: me.magnum.melonds.ui.settings.fragments.g
                    public final /* synthetic */ GeneralPreferencesFragment B;

                    {
                        this.B = this;
                    }

                    @Override // androidx.preference.p
                    public final void a(Preference preference) {
                        switch (r2) {
                            case 0:
                                this.B.f9593e0.a(null);
                                return;
                            default:
                                this.B.f9594f0.a(null);
                                return;
                        }
                    }
                });
                return;
            }
            return;
        }
        nc.k.f("rewindPreference");
        throw null;
    }

    @Override // androidx.fragment.app.j0
    public final void onResume() {
        super.onResume();
        MasterSwitchPreference masterSwitchPreference = this.f9592d0;
        Boolean bool = null;
        if (masterSwitchPreference != null) {
            o onPreferenceChangeListener = masterSwitchPreference.getOnPreferenceChangeListener();
            if (onPreferenceChangeListener != null) {
                MasterSwitchPreference masterSwitchPreference2 = this.f9592d0;
                if (masterSwitchPreference2 != null) {
                    if (masterSwitchPreference2 != null) {
                        SharedPreferences sharedPreferences = masterSwitchPreference2.getSharedPreferences();
                        if (sharedPreferences != null) {
                            MasterSwitchPreference masterSwitchPreference3 = this.f9592d0;
                            if (masterSwitchPreference3 != null) {
                                bool = Boolean.valueOf(sharedPreferences.getBoolean(masterSwitchPreference3.getKey(), false));
                            } else {
                                nc.k.f("rewindPreference");
                                throw null;
                            }
                        }
                        onPreferenceChangeListener.a(masterSwitchPreference2, bool);
                        return;
                    }
                    nc.k.f("rewindPreference");
                    throw null;
                }
                nc.k.f("rewindPreference");
                throw null;
            }
            return;
        }
        nc.k.f("rewindPreference");
        throw null;
    }
}

File: app/src/main/java/me/magnum/melonds/ui/settings/fragments/GeneralPreferencesFragment.kt
package me.magnum.melonds.ui.settings.fragments

import android.os.Bundle
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AlertDialog
import androidx.lifecycle.lifecycleScope
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceManager
import androidx.preference.SwitchPreference
import com.smp.masterswitchpreference.MasterSwitchPreference
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import me.magnum.melonds.R
import me.magnum.melonds.common.DirectoryAccessValidator
import me.magnum.melonds.common.UriPermissionManager
import me.magnum.melonds.extensions.isSustainedPerformanceModeAvailable
import me.magnum.melonds.impl.SettingsBackupManager
import me.magnum.melonds.ui.settings.PreferenceFragmentHelper
import me.magnum.melonds.ui.settings.PreferenceFragmentTitleProvider
import javax.inject.Inject

@AndroidEntryPoint
class GeneralPreferencesFragment : BasePreferenceFragment(), PreferenceFragmentTitleProvider {

    private val helper by lazy { PreferenceFragmentHelper(this, uriPermissionManager, directoryAccessValidator) }
    @Inject lateinit var uriPermissionManager: UriPermissionManager
    @Inject lateinit var directoryAccessValidator: DirectoryAccessValidator
    @Inject lateinit var settingsBackupManager: SettingsBackupManager

    private var rewindPreference: Preference? = null
    private lateinit var frameLimitSpeedPreference: ListPreference

    private val backupLauncher = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
        if (uri != null) {
            lifecycleScope.launch(Dispatchers.IO) {
                runCatching { settingsBackupManager.backup(uri) }
                    .onSuccess {
                        withContext(Dispatchers.Main) {
                            AlertDialog.Builder(requireContext())
                                .setMessage(R.string.settings_backup_success)
                                .setPositiveButton(android.R.string.ok, null)
                                .show()
                        }
                    }
                    .onFailure {
                        withContext(Dispatchers.Main) {
                            Toast.makeText(requireContext(), R.string.settings_backup_error, Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    private val restoreLauncher = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
        if (uri != null) {
            lifecycleScope.launch(Dispatchers.IO) {
                runCatching { settingsBackupManager.restore(uri) }
                    .onSuccess {
                        withContext(Dispatchers.Main) {
                            AlertDialog.Builder(requireContext())
                                .setMessage(R.string.settings_restore_success)
                                .setPositiveButton(android.R.string.ok, null)
                                .show()
                        }
                    }
                    .onFailure {
                        withContext(Dispatchers.Main) {
                            Toast.makeText(requireContext(), R.string.settings_restore_error, Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        runCatching { setPreferencesFromResource(R.xml.pref_general, rootKey) }
        runCatching { addPreferencesFromResource(R.xml.pref_general_updates) }

        findPreference<Preference>("enable_rewind")?.let { pref ->
            rewindPreference = pref
            pref.setOnPreferenceClickListener {
                me.magnum.melonds.ui.settings.dialogs.showSettingsRewindDialog(requireContext()) {
                    updateRewindSummary()
                }
                true
            }
            updateRewindSummary()
        }
        findPreference<ListPreference>("frame_limit_speed_multiplier")?.let { pref ->
            frameLimitSpeedPreference = pref
            helper.bindPreferenceSummaryToValue(pref)
            pref.sharedPreferences?.registerOnSharedPreferenceChangeListener(sharedPreferenceChangeListener)
        }
        findPreference<SwitchPreference>("enable_sustained_performance")?.let { pref ->
            pref.isVisible = requireContext().isSustainedPerformanceModeAvailable()
        }

        val themePreference = findPreference<ListPreference>("theme")
        if (themePreference != null) {
            helper.bindPreferenceSummaryToValue(themePreference)
            themePreference.setOnPreferenceChangeListener { _, newValue ->
                me.magnum.melonds.ui.theme.AppThemeManager.updateTheme(requireContext(), newValue as String)
                activity?.recreate()
                true
            }
        }

        val accentPreference = findPreference<ListPreference>("theme_accent_color")
        if (accentPreference != null) {
            if (accentPreference.value.isNullOrEmpty()) {
                accentPreference.value = "electric_cyan"
            }
            helper.bindPreferenceSummaryToValue(accentPreference)
            accentPreference.setOnPreferenceChangeListener { _, newValue ->
                val newAccent = newValue as String
                me.magnum.melonds.ui.theme.AppThemeManager.accentColorState.value = newAccent
                accentPreference.sharedPreferences?.edit()?.putString("theme_accent_color", newAccent)?.apply()
                view?.post { applyAccentColorToViews() }
                activity?.recreate()
                true
            }
        }

        val cardStylePreference = findPreference<ListPreference>("rom_card_style")
        if (cardStylePreference != null) {
            helper.bindPreferenceSummaryToValue(cardStylePreference)
            cardStylePreference.setOnPreferenceChangeListener { _, newValue ->
                val newStyle = newValue as String
                me.magnum.melonds.ui.theme.AppThemeManager.cardStyleState.value = newStyle
                cardStylePreference.sharedPreferences?.edit()?.putString("rom_card_style", newStyle)?.apply()
                true
            }
        }

        updateFrameLimitSpeedPreferenceState()

        findPreference<Preference>("backup_settings")?.setOnPreferenceClickListener {
            backupLauncher.launch(null)
            true
        }
        findPreference<Preference>("restore_settings")?.setOnPreferenceClickListener {
            restoreLauncher.launch(null)
            true
        }
    }

    override fun onResume() {
        super.onResume()
        updateRewindSummary()
        updateFrameLimitSpeedPreferenceState()
    }

    private fun updateRewindSummary() {
        val pref = rewindPreference ?: return
        val sp = androidx.preference.PreferenceManager.getDefaultSharedPreferences(requireContext())
        val isEnabled = sp.getBoolean("enable_rewind", false)
        val period = sp.getInt("rewind_period", 10)
        val window = sp.getInt("rewind_window", 6) * 10
        pref.summary = if (isEnabled) {
            "Включено (Интервал: ${period}с, Буфер: ${window}с)"
        } else {
            "Отключено"
        }
    }

    override fun onDestroy() {
        if (::frameLimitSpeedPreference.isInitialized) {
            frameLimitSpeedPreference.sharedPreferences?.unregisterOnSharedPreferenceChangeListener(sharedPreferenceChangeListener)
        }
        super.onDestroy()
    }

    override fun getTitle() = getString(R.string.category_general)

    private val sharedPreferenceChangeListener = android.content.SharedPreferences.OnSharedPreferenceChangeListener { _, key ->
        if (::frameLimitSpeedPreference.isInitialized && (key == "ra_hardcore_enabled" || key == frameLimitSpeedPreference.key)) {
            updateFrameLimitSpeedPreferenceState()
        }
    }

    private fun updateFrameLimitSpeedPreferenceState() {
        if (!::frameLimitSpeedPreference.isInitialized) return
        val preferences = PreferenceManager.getDefaultSharedPreferences(requireContext())
        val hardcoreEnabled = preferences.getBoolean("ra_hardcore_enabled", false)
        frameLimitSpeedPreference.isVisible = !hardcoreEnabled
        if (!hardcoreEnabled) {
            frameLimitSpeedPreference.summary = frameLimitSpeedPreference.entry ?: getString(R.string.not_set)
        }
    }
}
